dxr/plugins/xbl/analyzer.py

```diff
@@ -0,0 +1,78 @@
+from warnings import warn
+import xml.parsers.expat as expat
+
+from dxr.indexers import Extent, Position
+from dxr.plugins.xbl.refs import PLUGIN_NAME, TypeRef
+
+
+class XBLAnalyzer(object):
+    def __init__(self, tree, contents, encoding):
+        self.tree = tree
+        self.contents = contents
+        self.lines = contents.splitlines(True)
+        self.parser = expat.ParserCreate(encoding)
+        # Stack of namespaces, issued by the 'id' attr of bindings.
+        self.namespace = []
+        self.needles = []
+        self.refs = []
+        # Late bind our custom handlers to the parser
+        self.parser.StartElementHandler = self.StartElementHandler
+        self.parser.EndElementHandler = self.EndElementHandler
+        try:
+            self.parser.Parse(self.contents)
+        except expat.ExpatError:
+            warn("Exception occurred on parsing XBL file {}".format(self.path))
+
+    def extent_for_name(self, name):
+        """Return Extent for the next occurrence of name."""
+        # Because the parser does not provide location info on attr positions,
+        # we can only search for them from the current position.
+        found = False
+        for (number, line) in enumerate(self.lines[self.parser.CurrentLineNumber - 1:]):
+            if name in line:
+                row, col = self.parser.CurrentLineNumber + number, line.index(name)
```

Comment:
Seems kind of shoddy. Wouldn't a property called "property" find the tag name instead of the attr? To do this right, we'd have to reparse the start tag. I wonder if a different XML parser would return offsets for us.